crates/autopilot/src/domain/fee/mod.rs

```diff
@@ -23,15 +26,39 @@ use {
 /// Constructs fee policies based on the current configuration.
 pub struct ProtocolFee {
     policy: policy::Policy,
+    max_partner_fee: FeeFactor,
 }
 
-impl ProtocolFee {
-    pub fn new(fee_policy_args: arguments::FeePolicy) -> Self {
+impl From<arguments::FeePolicy> for ProtocolFee {
+    fn from(policy_arg: arguments::FeePolicy) -> Self {
```

Comment:
Is it necessary to move this function here and also make fields crate-public?

Reply:
sure!